### PR TITLE
[move] Fixes for enums in Move model + add support in docgen

### PR DESCRIPTION
**Pull request by tzakian**
## Description 

Fixes some issues with enums in Move model creation, and adds support for enums in docgen, and adding doc comments to enum variants. 

## Test plan 

Added a test for docgen with enums -- this tests both the bugfixes and also the generation of doc comments for enums. 

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

File: external-crates/move/crates/move-docgen/src/docgen.rs [hunk 1]
```diff
@@ -584,6 +584,15 @@ impl<'env> Docgen<'env> {
...
         }
 
+        if !module_env.get_enums().count() > 0 {
+            for s in module_env
+                .get_enums()
+                .sorted_by(|a, b| Ord::cmp(&a.get_loc(), &b.get_loc()))
+            {
+                self.gen_enum(&s);
+            }
+        }
```
Comment (cgswords) [inline external-crates/move/crates/move-docgen/src/docgen.rs:594]: It's weird to me that this checks the count instead of just doing the iteration, but the rest of the block is written that way, too.

File: external-crates/move/crates/move-model/src/builder/model_builder.rs [hunk 2]
```diff
@@ -27,26 +27,37 @@ use crate::{
...
     pub fun_table: BTreeMap<QualifiedSymbol, FunEntry>,
     /// A symbol table for constants.
     pub const_table: BTreeMap<QualifiedSymbol, ConstEntry>,
 }
 
-/// A declaration of a struct.
+/// A declaration of a datatype.
 #[derive(Debug, Clone)]
-pub(crate) struct StructEntry {
+pub(crate) struct DatatypeEntry {
```
Comment (cgswords) [inline external-crates/move/crates/move-model/src/builder/model_builder.rs:43]: Not this PR, but it would be nice to support positional datatypes at some point.

**Review by cgswords — APPROVED** [review]
LGTM. I think it could use a few more tests (make sure it doesn't break on empty enum definitions, etc) but overall it seems like a great fix-up until we can get Model 2 everywhere.
